workflow/controller/exec_control.go

```diff
@@ -27,7 +27,7 @@ func (woc *wfOperationCtx) applyExecutionControl(ctx context.Context, pod *apiv1
 		return nil
 	case apiv1.PodPending:
 		// Check if we are currently shutting down
-		if woc.execWf.Spec.Shutdown != "" {
+		if woc.wf.Spec.Shutdown != "" {
```

Comment:
This would be really easy to make mistakes. I know that there are notes on the use of `wf` v.s. `execWf` but I wonder if we could refactor a bit to prevent the misuse of `woc.execWf` on `Suspend` and `Shutdown`.

Reply:
Yes, Mainly `Shutdown` and `Suspend` can be changed during the runtime.  The above fix is only for the `WorkflowTemplateRef` scenario rest of the scenario will work as expected.
Possible refactoring for prevent:
1. Code can update `Shutdown` and `Suspend ` on workflow.spec` and `workflow.status.storedWorkflowSpec`.
2. Code needs to construct the `execWf` in each execution. This is a little bit costly operation on each execution. 

I am thinking option 1, We just use `execWf` for all places. Let me talk to the team about what they think.

Reply:
After given thought. if the user did patch through `kubectl` then option 1 won't work.